.github/workflows/build-irods-http-api-ubuntu.yml

```diff
@@ -0,0 +1,54 @@
+name: iRODS HTTP API Builder
```

Comment:
Looking at irods/irods, how do you feel about changing this to one of the following?
- `Build-Ubuntu`
- `build-irods-http-api-ubuntu`

Then we could expand this to cover other versions of Ubuntu, and later other platforms such as EL8 and EL9.

Reply:
I think it would be better without the hyphens. This name is supposed to be a user-facing name. You can see this in the passed checks at the bottom.

Perhaps we can name it so it ends up as `Ubuntu iRODS HTTP API Builder / Build iRODS HTTP API - <Specific Ubuntu Image Here>`?

Reply:
Sure. I don't think hyphens make it less readable.

Try expanding the builder workflow to support ub22/24, EL8/9 and see how the github actions report it. We want things to be sorted, grouped, and organized.

Aim for consistency across repos.

Take a look at the workflows for irods/irods.

Reply:
> Try expanding the builder workflow to support ub22/24, EL8/9 and see how the github actions report it. We want things to be sorted, grouped, and organized.

Expanded Ubuntu side.

Reply:
> I like the inclusion of iRODS 4.3.x in the entries, but it implies each release of iRODS will have an entry

Yes, each release would have an entry since we state a minimum only.

> unless we come up with a policy for when a dev package should be added/removed.

I'm not sure if I follow, are you asking what if we have a change of requirements or a breaking change?
~If our requirements change, then our tests would match that too, whenever the build container changes, right?~
In terms of iRODS version, isn't this just when our README or build container changes?

> I don't think that's necessary so let's stick with building against the minimum version of the iRODS dev package (i.e. 4.3.2). That aligns with the dev package used to produce the Docker images and it covers backwards compatibility.

We should test the minimum, but I think it would be nice to make sure there are no problems compiling with a more recent version (and eventually test against this too). There's also the concern of accidental forward "departures from interface" or breakages.

Reply:
> I'm not sure if I follow, are you asking what if we have a change of requirements or a breaking change?
~If our requirements change, then our tests would match that too, whenever the build container changes, right?~
In terms of iRODS version, isn't this just when our README or build container changes?

Consider the case where iRODS 4.3.7 is released and 4.3.2 is still the minimum version required for compiling the software.

While it would be nice to build against all versions, it's probably unrealistic to do that in GitHub due to potential restrictions/constraints. We'd need to define rules (a policy) for picking which versions to build against, else people may wonder **why** we build against a subset of the versions instead of all of them.

> We should test the minimum, but I think it would be nice to make sure there are no problems compiling with a more recent version (and eventually test against this too). There's also the concern of accidental forward "departures from interface" or breakages.

I agree. That's something that can be done by the k8s testing environment.

Keep in mind we'll continue to iterate on this. This is a good first step.

Reply:
In Discuss:

We will implement the other non Ubuntu images in a separate PR.

Reply:
Updated names.